libraries/AP_Math/quaternion.cpp

```diff
@@ -620,6 +620,33 @@ Quaternion Quaternion::operator*(const Quaternion &v) const
     return ret;
 }
 
+// Optimized quaternion rotation operator, equivalent to converting
+// (*this) to a rotation matrix then multiplying it to the argument `v`.
+//
+// 15 multiplies and 15 add / subtracts. Caches 3 floats
+Vector3f Quaternion::operator*(const Vector3f &v) const
```

Comment:
how is this different from the existing rotate() method?
https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Math/quaternion.cpp#L458